### PR TITLE
Modify event on firebase

### PR DESCRIPTION
**Pull request by idosavion**
Add the necessary function to Database Streamer and  started modifying view single event activity

### CODE REVIEW

File: app/src/main/java/com/huji/foodtricks/buddies/SignInActivity.java [hunk 1]
```diff
@@ -41,6 +41,8 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_signin);
+        Intent signedInIntent = new Intent(this, EventsTabsActivity.class);
```
Comment (mharsat) [inline app/src/main/java/com/huji/foodtricks/buddies/SignInActivity.java]: We already removed this because sign in works from all devices

**Review by mharsat — CHANGES_REQUESTED** [review]
All good just read the comment

File: app/src/main/java/com/huji/foodtricks/buddies/DatabaseStreamer.java [hunk 2]
```diff
@@ -161,18 +170,34 @@ public void onError(DatabaseError error) {
...
+                    @Override
+                    public void onSuccess(Void aVoid) {
+                        completion.onResponse();
+                    }
+                });
+    }
+
+
+
+    public void modifyEvent(EventModel eventModel, String eventModelId,
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/buddies/DatabaseStreamer.java]: can you rename `(EventModel eventModel)` to `modifiedEventModel` or `updatedEventModel`, just os it's clear that it's the new one and not the old one?

File: app/src/main/java/com/huji/foodtricks/buddies/ViewSingleEventActivity.java [hunk 3]
```diff
@@ -48,13 +52,35 @@ public boolean onPrepareMenu(NavigationMenu navigationMenu) {
             public boolean onMenuItemSelected(MenuItem menuItem) {
                 Toast.makeText(ViewSingleEventActivity.this, getString(R.string.change_rsvp_msg_prefix) + menuItem.getTitle(), Toast.LENGTH_SHORT).show();
                 EventAttendanceProvider attendanceProvider = curr_event.getAttendanceProvider();
+                String uId;
+                try {
+                    uId = FirebaseAuth.getInstance().getCurrentUser().getUid(); // Should use getIntent().getSerializable
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/buddies/ViewSingleEventActivity.java]: Just to make sure, uId is the UserModel's key in the database, not the Authentication key.

**Review by mfocshaner — COMMENTED** [review]
Looks really cool!
visual changes needed for next iteration (next PR, or this one if you have time):
- "Going?" is clickable, should be disabled
- background of buttons is white and not gray like the background; we need to ask the design team if it's ok.
- Once a button has been clicked, I think it should change its color.

Good job! <3

File: app/src/main/java/com/huji/foodtricks/buddies/SignInActivity.java (hunk 1 above)
Comment (idosavion) [reply app/src/main/java/com/huji/foodtricks/buddies/SignInActivity.java]: fixed

File: app/src/main/java/com/huji/foodtricks/buddies/DatabaseStreamer.java (hunk 2 above)
Comment (idosavion) [reply app/src/main/java/com/huji/foodtricks/buddies/DatabaseStreamer.java]: renamed

File: app/src/main/java/com/huji/foodtricks/buddies/ViewSingleEventActivity.java (hunk 3 above)
Comment (idosavion) [reply app/src/main/java/com/huji/foodtricks/buddies/ViewSingleEventActivity.java]: yes

**Comment by idosavion** [comment]
All fixed,
about the buttons in ViewSingleEventActivity: the activity is not finished but to avoid conflicts I want to merge the significant changes as fast as possible

File: app/src/main/java/com/huji/foodtricks/buddies/ViewSingleEventActivity.java [hunk 4]
```diff
@@ -136,10 +136,12 @@ public void onError(DatabaseError error) {
         allRSVPButtons.remove(view.getId()); // remove the selected button from the list of buttons to disable
         Button selectedButtonView = (Button) view; // just for readability
         selectedButtonView.setBackgroundColor(Color.MAGENTA);
+        selectedButtonView.setTextColor(Color.WHITE);
 //        setContentView(R.layout.activity_view_single_event);
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/buddies/ViewSingleEventActivity.java:141]: remove comment, add:
`selectedButtonView.setEnabled(false);`

**Review by mharsat — APPROVED** [review]

**Pull request merged**